### DISCUSSION TITLE
Webhook is not triggered for changing LAG in an Interface

### DISCUSSION THREAD

**Discussion by jayantdhole37** [nautobot/nautobot#6168] [Q&A]
Hello Everyone,
I recently started using nautobot to automate our  Switches fleet. The General Workflow of the automation is as follows:
You make changes in Nautobot -> nautobot detects changes (whether its create , update , delete ) and then it triggers a webhook to ansible with all the information needed -> and ansible takes care of the configuration changes.

i am having problems with LAG interface. i am able to create , update and delete a LAG interface and even able to assign the LAG to an individual interface.
The problem arises and will explain with an example. 
lets say I have created a Lag interface (ae0) and assigned that lag ae0 to ge-0/0/0 interface. 

When I delete the lag interface (ae0) nautobot automatically triggers a webhook looking at the delete action for ae0 and goes ahead start the ansible playbook.
When I delete ae0 , in the interface ge-0/0/0 the lag field is removed as well which is expected. but since the ge-0/0/0 field is changed as well. It should ideally trigger a webhook for ge-0/0/0 (update or anything) to push the configuration as this interface  is updated as well.

whenever i delete the lag interface, I am able to event changes for LAG but nothing for the interface. 

i am not sure whether this is a bug or is this behavior expected. Any help is appreciated and please do let me know if any additional information is required
If not this. what is the recommended method to do changes related to LAG and its interfaces. 

Even when I check Individual Interface Options. It dies not have LAG options, which makes me believe that Individual interfaces are not actually pulling the LAG information hence no change is detected. 
![image](https://github.com/user-attachments/assets/5fa13080-b88f-4149-8100-cba6304837b6)

This section has all the information except LAG.

**Comment by abates** [comment]
I'll take a stab at this:

The `lag` field on the `Interfaces` model (which is what is used to build the `member_interfaces` relationship) is a `ForeignKey`. What this means is that there is no actual field containing the member interfaces. Instead, when an interface is displayed, a query is performed to see if any member interfaces exist. If a member interface is deleted, the specific member interface record in the database is deleted and the webhook is fired (which you are seeing). However, the parent interface is actually not changed. Therefore, I think no webhook being called for the parent interface is actually expected behavior.

Perhaps a workaround is to simply have an API callback to Nautobot from the playbook to get a list of the current member interfaces and always update the parent if a member has been removed?
